### PR TITLE
ENH: stats: add 'ddof' parameter to the 'variation' function

### LINKED ISSUE #13341: add scipy.stats.variation with an ddof parameter
**Is your feature request related to a problem? Please describe.**
Good afternoon dear developer community,

in my opinion it would make sense to include a ddof parameter, to adjust the standard deviation of the inbuild numpy std, that supports the ddof statement.

The documentation states that it calculates the variation of coefficient in a biased case. 

grafik

This is only true for sample statistics, for Normal distributed population statistics, the calculation procedure is correct and not biased. Therefore the documentation should be specified more precisely. 

**Describe the solution you'd like**

I would add the ddof parameter as an additional parameter with a default value of 0 to ensure compatibility. the user can adjust the values based it his/her needs.

proposed_variation(a, axis=0, nan_policy='propagate', **ddof=0**)
...
...

return a.std(axis,ddof=ddof) / a.mean(axis) # change the last line. Same needs to applied into mstats.variation as the only difference is not supporting masked values

**Describe alternatives you've considered**

Calculate the coefficient of variation by using the "handwritten" formula

import numpy as np
np.std(a,ddof=1)/np.mean(a) 

**Additional context (e.g. screenshots)**

grafik

Thank you very much for your efforts.

### PR DESCRIPTION
**Pull request by tirthasheshpatel**
#### Reference issue
closes gh-13341

#### What does this implement/fix?
A 'ddof' parameter has been added to the 'variation' function in the 'stats' and 'mstats' module to let the user specify the delta degrees of freedom for the calculation of CV. This is useful, for instance, in the calculation of sample CV by setting 'ddof=1', hence, using the unbiased variance of the sample. Regression tests for it have also been added.

### CODE REVIEW

**Review by smurfit89 — APPROVED** [review]
Looks good

**Review by WarrenWeckesser — APPROVED** [review]
Thanks @tirthasheshpatel, this looks good.

One suggestion, but not necessary for merging this PR, is to improve the docstring.  We could replace

    Computes the coefficient of variation, the ratio of the biased standard
    deviation to the mean.

with

    Compute the coefficient of variation.

    The coefficient of variation is the standard deviation divided by the
    mean.  This function is equivalent to:

       np.std(x, axis=axis, ddof=ddof) / np.mean(x)

    The default for ``ddof`` is 0, but many definitions of the coefficient
    of variation use the square root of the unbiased sample variance
    for the sample standard deviation, which corresponds to ``ddof=1``.

What do you think?

**Comment by tirthasheshpatel** [comment]
> What do you think?

I absolutely agree, @WarrenWeckesser. Thanks for the suggestion!

File: scipy/stats/mstats_basic.py [hunk 1]
```diff
@@ -2219,10 +2219,18 @@ def moment(a, moment=1, axis=0):
...
+def variation(a, axis=0, ddof=0):
     """
-    Computes the coefficient of variation, the ratio of the biased standard
-    deviation to the mean.
+    Compute the coefficient of variation.
+
+    The coefficient of variation is the standard deviation divided by the
+    mean.  This function is equivalent to:
+
+    ``np.std(x, axis=axis, ddof=ddof) / np.mean(x)``
```
Comment (WarrenWeckesser) [inline scipy/stats/mstats_basic.py:2229]: Tweak markup:
```suggestion
    mean.  This function is equivalent to::

        np.std(x, axis=axis, ddof=ddof) / np.mean(x)
```

File: scipy/stats/stats.py [hunk 2]
```diff
@@ -1093,12 +1093,18 @@ def _moment(a, moment, axis):
...
+def variation(a, axis=0, nan_policy='propagate', ddof=0):
     """
     Compute the coefficient of variation.
 
-    The coefficient of variation is the ratio of the biased standard
-    deviation to the mean.
+    The coefficient of variation is the standard deviation divided by the
+    mean.  This function is equivalent to:
+
+    ``np.std(x, axis=axis, ddof=ddof) / np.mean(x)``
```
Comment (WarrenWeckesser) [inline scipy/stats/stats.py:1103]:
```suggestion
    mean.  This function is equivalent to::

        np.std(x, axis=axis, ddof=ddof) / np.mean(x)
```

**Review by WarrenWeckesser — APPROVED** [review]
Thanks @tirthasheshpatel, the update looks good.

I'll make a couple tweaks to the markup, and merge.

File: scipy/stats/stats.py [hunk 3]
```diff
@@ -1114,6 +1120,8 @@ def variation(a, axis=0, nan_policy='propagate'):
           * 'propagate': returns nan
           * 'raise': throws an error
           * 'omit': performs the calculations ignoring nan values
```
Comment (WarrenWeckesser) [inline scipy/stats/stats.py:1122]: Another tweak of the markup:
```suggestion
        * 'propagate': returns nan
        * 'raise': throws an error
        * 'omit': performs the calculations ignoring nan values

```

**Comment by WarrenWeckesser** [comment]
Thanks @tirthasheshpatel.  I added a short note about this to the [release notes](https://github.com/scipy/scipy/wiki/Release-note-entries-for-SciPy-1.7.0) on the wiki.

**Pull request merged**